File: 35-search-insert-position/search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        def binarySearch(arr,x,l,r):    
            while(l<=r):
                mid = (l+r) // 2
                if(arr[mid] == x):
                    return mid
                elif(arr[mid] < x):
                    l = mid + 1
                else:
                    r = mid - 1
            return l
        n = len(nums)
        result = binarySearch(nums,target,0,n-1)
        return result
        
        

File: 35-search-insert-position/search-insert-position.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    let l = 0, r = nums.length

    while(l < r){
        let mid = l + Math.floor((r-l)/2)

        if(nums[mid] == target) return mid
        else if(nums[mid] < target) l = mid + 1
        else r = mid
    }
    return l
};